### ISSUE TITLE
Identity swaps in maDLC 3.0+ with id=True

### ISSUE THREAD

**Issue by SGasperoni** [DeepLabCut/DeepLabCut#2867]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug description

Hi! I have been training a model with the new version of DLC. In my set up, I have 4 mice that are clearly identifiable (4 different colors). When I train the model and check it against the labelled frames, performance seems quite good. However, upon creating labeled videos, I noticed huge swaps between mice idenitity. These occur especially when the mice are in close proximity (despite focusing on frames where interactions occur during the frame extraction/labelling phase). Example: if mouse A and B are approaching each other, most of the time A will be incorrectly recognized as B, and viceversa. However, there are also several instances where A and B are approaching, mouse C is on the opposite side or very far, and A will be recognized as C, B as A, and C as B. Whenever all mice are far from each other tracking is good.
For context, I have trained with 500+ labelled frames. I have tried both Pytorch and TF engines, and it leads to the same problem. This is also not improved if I try to refine tracklets.
I went back to older versions of DLC I was using (DLC 2.3.3 and 2.3.5), and training with the same frames and shuffle, gave me good results, so it seems to be a specific problem of version 3+.

Thanks ☺️

### Operating System

Windows

### DeepLabCut version

3.0.0rc6

### DeepLabCut mode

multi animal

### Device type

NVIDIA GeForce RTX 4090

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
hey @Sere-98! Thanks for reporting this! Just to be sure, in your project `config.yaml` do you set:

```yaml
# Project definitions (do not edit)
...
identity: true
```

Could you also share the results you get when calling `evaluate_network` with your PyTorch model? And the exact call you make to `analyze_videos`?

**Comment by SGasperoni** [comment]
Hi, yes I have set identity: true in the config.yaml.
To analyze videos, I tried several options. Usually the one that worked for me with previous versions was:

`deeplabcut.analayze_videos(config_path, video_path, auto_track=False, identity_only=True)
deeplabcut.convertdetections2tracklets(config_path, video_path, track_method='ellipse', identity_only=True)
deeplabcut.stitch_tracklets(config_path, video_path, split_tracklets=False, track_method='ellipse')`

I have attached the file with the results when I call evaluate_network. 

[CombinedEvaluation-results.csv](https://github.com/user-attachments/files/18841560/CombinedEvaluation-results.csv)

**Comment by n-poulsen** [comment]
Thanks for the info @Sere-98! I tested the identity head again on a dataset with a black and a white mouse and get perfect tracking results.

From the model performance you shared, your model should be performing well too (ID head test accuracy above 90% for all bodyparts but the tail tip).

It's very difficult to debug what's causing the performance issue without the actual data; I don't think I actually need any videos, as I should be able to work just from the assemblies. Would you be willing to share the `_assemblies.pickle` file that was produced by `analyze_videos`?

**Comment by SGasperoni** [comment]
Yes, I was also quite surprised to see low performance, given the results of the evaluation.
I tried attaching one of the assemblies.pickle file, but it doesn't let me do it. Any suggestion on what to convert it to? I tried .json but the upload also fails.

**Comment by n-poulsen** [comment]
You could also share it through a link to a Google drive (likely the easiest way to do it)

**Comment by SGasperoni** [comment]
Here it is:
https://drive.google.com/drive/folders/1T4UXDDxZtF2uaH7s7m-fI88A9kU3hzNU?usp=sharing

**Comment by n-poulsen** [comment]
Thanks for sharing the data @Sere-98 - I think I figured out where the issue came from and I'm working on fixing it! I'll let you know when it's done

**Comment by n-poulsen** [comment]
@Sere-98 this issue should have been fixed (or at least partly fixed) by #2903 - while there may still be issues in the quality of predictions, tracking performance with `identity_only=True` should be much better. Let me know if this fully fixes your issue!

Once the pull request is merged, you can update your code to include the changes with:

```bash
pip uninstall deeplabcut
pip install "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut"
```

**Comment by n-poulsen** [comment]
Closing as stale.

**Comment by SGasperoni** [comment]
Hi @n-poulsen ,
sorry for not replying earlier, I was out of office.
I have tested after the update; while there seems to be some improvement, the error largely persists.
At the same link as last time (https://drive.google.com/file/d/1SN_b8litXnPK_SYTLYXis9JGXkeCrsU-/view?usp=sharing), I have added a short videoclip obtained after the tracking, hope it helps.

**Comment by n-poulsen** [comment]
Thanks @Sere-98 - I'll continue looking into it then!

**Comment by MMathisLab** [comment]
Can you try our new BUCTD or RTMpose models? This will likely solve the issues noted

**Comment by felipearrudamoura** [comment]
Hi @MMathisLab Are the BUCTD or RTMpose models available to run in the GUI? Thank you!

**Comment by MMathisLab** [comment]
We don't have them in the gui, but we have demo notebooks to help you use them ❤️

**Comment by felipearrudamoura** [comment]
@MMathisLab great!! Thank you very much!!

**Comment by felipearrudamoura** [comment]
@MMathisLab I followed the demo notebook. Just checking if I understood properly: 
1) before training with BUCTD and RTMpose, I need to train the model in the tradictional way in deeplabcut;
2) With the images labeled and the training datasets, then I can follow the "Training BU model" step and then the " Training CTD model".

Is that correct, or I don't need to perform the first training?

Thank you!

**Comment by MMathisLab** [comment]
you can skip step 1, you just train a standard BU DLC model; see here! https://github.com/DeepLabCut/DeepLabCut/blob/main/examples/COLAB/COLAB_BUCTD_and_CTD_tracking.ipynb
